bundles/CoreBundle/src/Migrations/Version20230321133700.php

```diff
@@ -17,55 +17,95 @@
 
 namespace Pimcore\Bundle\CoreBundle\Migrations;
 
+use DateTime;
+use DateTimeZone;
 use Doctrine\DBAL\Schema\Schema;
 use Doctrine\Migrations\AbstractMigration;
+use Exception;
 
 final class Version20230321133700 extends AbstractMigration
 {
     public function getDescription(): string
     {
-        return 'Alters date time columns to timestamp columns for application logs, notifications and scheduled tasks. Converts values to UTC.';
+        return 'Converts datetime/timestamp values to UTC for application logs and notifications.';
     }
 
-    private function converToTimeZone(
+    private function convertToTimeZone(
         Schema $schema,
         string $table,
         string $timeStampColumn,
         bool $up = true
     ): void {
-        if ($schema->hasTable($table)) {
-            $db = \Pimcore\Db::get();
-            $fromTimeZone = $up ? date_default_timezone_get() : 'UTC';
-            $toTimeZone = $up ? 'UTC' : date_default_timezone_get();
-
-            $this->addSql(
-                sprintf(
-                    'update %s set %s = CONVERT_TZ(%s,%s,%s)',
-                    $db->quoteIdentifier($table),
-                    $db->quoteIdentifier($timeStampColumn),
-                    $db->quoteIdentifier($timeStampColumn),
-                    $db->quote($fromTimeZone),
-                    $db->quote($toTimeZone)
-                )
-            );
+        if (!$schema->hasTable($table)) {
+            return;
+        }
+
+        $db = \Pimcore\Db::get();
+        $fromTimeZone = $up ? date_default_timezone_get() : 'UTC';
+        $toTimeZone = $up ? 'UTC' : date_default_timezone_get();
+
+        // Test if MySQL CONVERT_TZ works properly
+        $testResult = $db->fetchOne(
+            "SELECT CONVERT_TZ('2000-01-01 00:00:00', ?, ?)",
+            [$fromTimeZone, $toTimeZone]
+        );
+
+        $usePhpFallback = !$testResult || $testResult === '0000-00-00 00:00:00';
+
+        if (!$usePhpFallback) {
+            $this->addSql(sprintf(
+                'UPDATE %s SET %s = CONVERT_TZ(%s, %s, %s)',
+                $db->quoteIdentifier($table),
+                $db->quoteIdentifier($timeStampColumn),
+                $db->quoteIdentifier($timeStampColumn),
+                $db->quote($fromTimeZone),
+                $db->quote($toTimeZone)
+            ));
+
+            return;
+        }
+
+        // ⚙️ Fallback to PHP-based conversion
+        $rows = $db->fetchAllAssociative(sprintf(
```

Comment:
We could iterate lazy using `yield` instead of fetching all rows into memory before iteration:

```suggestion
        $rows = $db->iterateAssociative(sprintf(
```